Move updater notices to notification API

DaddyBoard opened the pull request:
Current implementation as follows;
* Directly replaced the `Notices.show` for `Notifications.show` 
* A separate notification will spawn for BD Core, Plugin and Theme updates. 

Open to any rewording or styling changes.

![Discord_Z6m3Y4qOb2](https://github.com/user-attachments/assets/b8c1a1fa-70e8-44d4-8a8b-ae40e96feb97)

This iteration **requires functionality** from https://github.com/BetterDiscord/BetterDiscord/pull/1918. Should this other PR not be desirable, I can change it to something else np

<details>
<summary>OLD</summary>

Currently, I have setup the duration to be;
* **IF** automatic update checking is enabled; the duration of the notification will match the time their auto-check is set to. This will stop duplicate notifications popping when AFK. 
* **IF** automatic checking is disabled, infinity.

</details>

zerebos commented on src/betterdiscord/modules/updater.ts:
```diff
@@ -181,14 +181,17 @@ export class CoreUpdater {
+        Notifications.show({
+            title: t("Updater.updateAvailable", {version: this.remoteVersion}),
+            type: "info",
+            icon: () => React.createElement("img", {src: "https://betterdiscord.app/resources/branding/logo_small.svg", width: "18px", height: "18px"}),
```
Don't link a remote image, use the icon in the codebase.

zerebos commented on src/betterdiscord/modules/updater.ts:
```diff
@@ -318,14 +321,44 @@ class AddonUpdater {
+                ]).flat().slice(0, -1)
+            ],
+            type: "info",
+            icon: () => React.createElement("img", {src: "https://betterdiscord.app/resources/branding/logo_small.svg", width: "18px", height: "18px"}),
```
Don't remote link image

DaddyBoard replied:
Changed to
```js
icon: () => React.createElement("img", {
              src: "data:image/svg+xml;base64,PHN2ZyB2ZXJzaW9uPSIxLjEiIHhtbG5zPSJodHRwOi8vd3d3LnczLm9yZy8yMDAwL3N2ZyIgeG1sbnM6eGxpbms9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkveGxpbmsiIHg9IjBweCIgeT0iMHB4IiB2aWV3Qm94PSIwIDAgMjAwMCAyMDAwIiBlbmFibGUtYmFja2dyb3VuZD0ibmV3IDAgMCAyMDAwIDIwMDAiIHhtbDpzcGFjZT0icHJlc2VydmUiPjxnPjxwYXRoIGZpbGw9IiMzRTgyRTUiIGQ9Ik0xNDAyLjIsNjMxLjdjLTkuNy0zNTMuNC0yODYuMi00OTYtNjQyLjYtNDk2SDY4LjR2NzE0LjFsNDQyLDM5OFY0OTAuN2gyNTdjMjc0LjUsMCwyNzQuNSwzNDQuOSwwLDM0NC45SDU5Ny42djMyOS41aDE2OS44YzI3NC41LDAsMjc0LjUsMzQ0LjgsMCwzNDQuOGgtNjk5djM1NC45aDY5MS4yYzM1Ni4zLDAsNjMyLjgtMTQyLjYsNjQyLjYtNDk2YzAtMTYyLjYtNDQuNS0yODQuMS0xMjIuOS0zNjguNkMxMzU3LjcsOTE1LjgsMTQwMi4yLDc5NC4zLDE0MDIuMiw2MzEuN3oiLz48cGF0aCBmaWxsPSIjRkZGRkZGIiBkPSJNMTI2Mi41LDEzNS4yTDEyNjIuNSwxMzUuMmwtNzYuOCwwYzI2LjYsMTMuMyw1MS43LDI4LjEsNzUsNDQuM2M3MC43LDQ5LjEsMTI2LjEsMTExLjUsMTY0LjYsMTg1LjNjMzkuOSw3Ni42LDYxLjUsMTY1LjYsNjQuMywyNjQuNmwwLDEuMnYxLjJjMCwxNDEuMSwwLDU5Ni4xLDAsNzM3LjF2MS4ybDAsMS4yYy0yLjcsOTktMjQuMywxODgtNjQuMywyNjQuNmMtMzguNSw3My44LTkzLjgsMTM2LjItMTY0LjYsMTg1LjNjLTIyLjYsMTUuNy00Ni45LDMwLjEtNzIuNiw0My4xaDcyLjVjMzQ2LjIsMS45LDY3MS0xNzEuMiw2NzEtNTY3LjlWNzE2LjdDMTkzMy41LDMxMi4yLDE2MDguNywxMzUuMiwxMjYyLjUsMTM1LjJ6Ii8+PC9nPjwvc3ZnPg==",
              width: "18px", 
              height: "18px"
            }),
            ```

zerebos commented on src/betterdiscord/modules/updater.ts:
```diff
@@ -318,14 +321,44 @@ class AddonUpdater {
+            title: `Addon Updater`,
+            content: [
+                t("Updater.addonUpdatesAvailable", {count: this.pending.length, type: this.type}),
+                React.createElement("br"),
+                ...addonNames.map(addon => [
+                    `• ${addon.name} `,
+                    React.createElement("i", {}, `(${addon.version})`),
+                    React.createElement("br")
+                ]).flat().slice(0, -1)
```
It would make more sense to use idiomatic blocks like `ul` and `li` instead of a manual bullet point with `br` breaks

zerebos commented on src/betterdiscord/modules/updater.ts:
```diff
@@ -181,14 +181,17 @@ export class CoreUpdater {
-                    UserSettingsWindow?.open?.("updates");
+        Notifications.show({
+            title: t("Updater.updateAvailable", {version: this.remoteVersion}),
+            type: "info",
```
Might be worth distinguishing the core update from addon updates to show its importance

DaddyBoard replied:
Could you expand on what you mean here? 

![image](https://github.com/user-attachments/assets/193564be-7e56-4d82-95ae-a466cea7fed6)

They do already have their own notification

zerebos replied:
Distinguishing them with something like different colors. To me, core update is major importance, addon updates and info importance

DaddyBoard replied:
Ahh gotcha, easy fix

DaddyBoard replied:
I have set Core updates to `warning` to show as red. Red is typically found with importance. 

Kept plugins and themes as blue (`info`)

DaddyBoard commented:
Updated with changes requested by Zere.

Changed creation/duration behaviour to now work with functionality from #1918. Should this not be merged I can change it to something else np

DaddyBoard commented:
@doggybootsy updated as requested

zerebos commented on assets/locales/en-us.json:
```diff
@@ -357,9 +357,11 @@
         "addonUpdatesAvailable": {
-            "one": "BetterDiscord has found an update for {{count}} of your {{type}}!",
-            "other": "BetterDiscord has found updates for {{count}} of your {{type}}!"
+            "title": "Addon Updater",
```
`title` should not be under `addonUpdatesAvailable`. `addonUpdatesAvailable` is a pluralization and should only have pluralization children

DaddyBoard commented:
@zerebos done as requested

doggybootsy commented on src/betterdiscord/modules/updater.ts:
```diff
@@ -318,14 +326,49 @@ export class AddonUpdater {
+
+        Notifications.show({
+            id: `addon-updates-${this.type}`,
+            title: t("Updater.addonUpdatesAvailable.title"),
```
The i18n change you made isn't reflected here

DaddyBoard replied:
Saving files definitely helps...😭 my bad, didn't save it

DaddyBoard commented:
@doggybootsy updated, mb on that one
